Source/SwiftLintFramework/Rules/FunctionNamingCaseRule.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  FunctionNamingCaseRule.swift
+//  SwiftLint
+//
+//  Created by Paul Williamson on 18/12/2015.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public struct FunctionNamingCaseRule: Rule {
+    public static let description = RuleDescription(
+        identifier: "function_naming_case",
+        name: "Function Naming Case",
+        description: "Function names should begin with a lower case letter.",
+        nonTriggeringExamples: [
+            "func doSomething() { }"
+        ],
+        triggeringExamples: [
+            "func DoSomething() { }"
+        ]
+    )
+
+    public func validateFile(file: File) -> [StyleViolation] {
+        let pattern = "func\\s+[A-Z]"
+        let excludingKinds = SyntaxKind.commentAndStringKinds()
```

Comment:
See [FunctionBodyLengthRule](https://github.com/realm/SwiftLint/blob/0.5.1/Source/SwiftLintFramework/Rules/FunctionBodyLengthRule.swift#L35-L50) for an example that does this.